pkgs/development/tools/continuous-integration/concourse/assets/main/default.nix

```diff
@@ -1,36 +1,71 @@
 {
   cacert,
+  callPackage,
+  elmPackages,
   gmp,
-  mkYarnPackage,
+  gnused,
+  lib,
   nodePackages,
   nss,
   patchelf,
+  pkgs,
   src,
   stdenv,
   zlib,
 }:
-mkYarnPackage {
-  name = "concourse-main-assets";
-  nativeBuildInputs = [ nodePackages.yarn patchelf cacert ];
-  buildInputs = [ cacert ];
-  inherit src;
-  yarnNix = ./yarn.nix;
-  pkgConfig = {
-    elm = {
-      postInstall = ''
-        find . -type f -executable -exec patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" {} \;
-        find . -type f -executable -exec patchelf --set-rpath "${gmp}/lib:${zlib}/lib:${nss}/lib" {} \;
+let
+  mkElmPackage =
+    { srcs ? ./elm-srcs.nix
+    , src
+    , name
+    , srcdir ? "./src"
+    , targets ? []
+    , versionsDat ? ./versions.dat
+    }:
+    stdenv.mkDerivation {
+      inherit name src;
+
+      buildInputs = [ elmPackages.elm ];
+
+      buildPhase = elmPackages.fetchElmDeps {
+        elmPackages = import srcs;
+        inherit versionsDat;
+      };
+
+      installPhase = let
+        elmfile = module: "${srcdir}/${builtins.replaceStrings ["."] ["/"] module}.elm";
+      in ''
+        mkdir -p $out/share/doc
+        ${lib.concatStrings (map (module: ''
+          echo "compiling ${elmfile module}"
+          elm make ${elmfile module} --output $out/${module}.js
+        '') targets)}
       '';
     };
+  elm-package = mkElmPackage {
+    src = src + "/web/elm";
+    name = "concourse-web";
+    targets = [ "Main" ];
   };
-  yarnFlags = [
-    "--offline"
-    "--frozen-lockfile"
+  offline-cache = (callPackage ./yarn.nix {}).offline_cache;
+in
+stdenv.mkDerivation {
+  name = "concourse-main-assets";
+  nativeBuildInputs = [
+    gnused
+    nodePackages.uglify-js
+    nodePackages.yarn
+    patchelf
   ];
-  preBuild = ''
+  inherit src;
+  buildPhase = ''
```

Comment:
Are we still using yarn with mkElmPackage?

Reply:
Actually, not necessarily! :laughing: